urllib3/poolmanager.py

```diff
@@ -186,9 +199,9 @@ def connection_from_context(self, request_context):
         pool_key_constructor = self.key_fn_by_scheme[scheme]
         pool_key = pool_key_constructor(request_context)
 
-        return self.connection_from_pool_key(pool_key)
+        return self.connection_from_pool_key(pool_key, request_context=request_context)
```

Comment:
Yeah, to be clear, I think we should key on the entire context. I don't think there is anything in the context that shouldn't be part of the pool key.